package-res/ccc/core/base/text.js

```diff
@@ -24,58 +24,57 @@ pvc.text = {
         if(text === '') return text;
 
         var textLen = pv.Text.measureWidth(text, font);
-        if(textLen <= len) return text;
+        if(textLen > len)
+            text = pvc.text.trimToWidthBin(len, text, font, trimTerminator, before, clipLen);
 
-        // ----------------
-        // Trim needed
-        if(textLen > len * 1.5) // threshold for using other algorithm
-            return pvc.text.trimToWidthBin(len, text, font, trimTerminator, before, clipLen);
-
-        len = Math.max(0, len - pv.Text.measureWidth(trimTerminator, font));
-
-        while(textLen > len) {
-            text = before ? text.slice(1) : text.slice(0, text.length -1);
-            textLen = pv.Text.measureWidth(text, font);
-        }
-
-        // "A.."" -> ""
-        // "AB.." -> "AB.."
-        // "ABC.." -> "AB.."
-        if(clipLen && textLen <= clipLen) return "";
-
-        return before ? (trimTerminator + text) : (text + trimTerminator);
+        return text;
     },
 
     trimToWidthBin: function(len, text, font, trimTerminator, before, clipLen) {
+        var highLen = pv.Text.measureWidth(text, font);
+        if(highLen <= len) return text;
 
-        len = Math.max(0, len - pv.Text.measureWidth(trimTerminator, font));
+        var lowLen = 0;
 
-        var ilen = text.length,
-            high = ilen - 2,
-            low = 0,
-            mid,
-            textLen;
+        var targetLen = Math.max(0, len - pv.Text.measureWidth(trimTerminator, font));
```

Comment:
Actually, pv.Text calls are cached. That's why this line was added: https://github.com/webdetails/ccc/pull/255/files#diff-ea834ea74dd29679b7b5a5a4d794a2bfR60.